diff --git a/BaseAll b/BaseAll
index 47daba7..f4adbab 100755
--- a/BaseAll
+++ b/BaseAll
@@ -19,6 +19,10 @@ LinkProject()
 	# But synobios need kernel header
 	PROJECTS="lnxscripts source/${SynoBaseProj} source/lnxsdk source/synobios"
 
+	if [ -n "$SYNO_LNXSYNOBASE_PATH" ]; then
+		PROJECTS=$(echo "$PROJECTS" | sed s,source/${SynoBaseProj},,)
+	fi
+
 	mkdir -p "${DESTINATION}/source"
 	for dir in ${PROJECTS}
 	do
@@ -68,9 +72,24 @@ if [ ! -d ${BASE} -o ! -d ${BASE}/lnxscripts ]; then
 fi
 
 . ${BASE}/lnxscripts/include/config
+if [ -n "$SYNO_LNXSYNOBASE_PATH" ]; then
+	if [ x"$BASEDIR" = x"/synosrc/curr" ]; then
+	# FIXME this could only work for update curr
+	# switch the branch to same as lnxscript
+		echo "Updating $BasePath first..."
+		( cd $BasePath || { echo "failed"; exit 1; }
+		  git pull
+		  cd -
+		)
+	else 
+		echo "not my branch, do not update lnxsynobase..."
+	fi
+	${BASE}/lnxscripts/SynoUpdate lnxsdk synobios
+else
 echo "Updating ${BASE}/source/${SynoBaseProj} first..."
 echo ""
 ${BASE}/lnxscripts/SynoUpdate ${SynoBaseProj} lnxsdk synobios
+fi
 
 if [ "$DoesRemoveOld" = "" ]; then
 	printf "Should I remove old [bin usr sbin lib var tmp etc dev opt] ? (Y/N) [Y]: "
diff --git a/SynoBase b/SynoBase
index db29ea9..a4740cb 100755
--- a/SynoBase
+++ b/SynoBase
@@ -13,7 +13,11 @@ fi
 . $ScriptsDir/include/platforms
 
 BasePath="${SourceDir}/$SynoBaseProj"
-if [ ! -d "${BasePath}" ]; then
+if [ -n "$SYNO_LNXSYNOBASE_PATH" ]; then
+	# NOTE: could overwrite BasePath from env
+	BasePath="$SYNO_LNXSYNOBASE_PATH"
+fi
+if [ ! -d "${BasePath}" -a ! -n "$SYNO_LNXSYNOBASE_PATH" ]; then
 	echo "SynoBaseProj(${BasePath}) project does not exist. Checking out first..."
 	# workaround for VERSION file temporarily
 	$ScriptsDir/SynoUpdate $SynoBaseProj lnxsdk synobios
diff --git a/SynoModel b/SynoModel
index 77210d5..e460677 100755
--- a/SynoModel
+++ b/SynoModel
@@ -813,6 +813,10 @@ CopyBuiltinPackage() {
 		echo skip copy built-in package when build flash only
 		return 0
 	fi
+	if ! BuildMachineOnly; then
+		# i don't want built-in packages
+		PackageBuiltin=
+	fi
 	local Package_Files=
 	local package_total_size=0
 	echo "Pack builtin packages: " $PackageBuiltin
diff --git a/include/config b/include/config
index 389dad2..9bf1565 100644
--- a/include/config
+++ b/include/config
@@ -729,7 +729,6 @@ CheckOutList="
 	ctdb-2.5.x
 	locale
 	${BuildList}
-	lnxsynobase
 	hotplug20030805-virtual-junior
 	valgrind
 	uistring
